typescript/widgets/src/chains/ChainSearchMenu.tsx

```diff
@@ -0,0 +1,227 @@
+    ),
+    [ChainListItem, customListItemField],
+  );
+
+  if (drilldownChain && chainMetadata[drilldownChain]) {
+    return (
+      <ChainDetailsMenu
+        chainMetadata={chainMetadata[drilldownChain]}
+        onClickBack={() => setDrilldownChain(undefined)}
+      />
+    );
+  } else {
+    return (
+      <SearchMenu<
+        ChainMetadata<{ disabled?: boolean }>,
+        ChainSortAndFilterState
+      >
+        data={data}
+        searchFn={chainSearch}
+        onClickItem={onClickChain}
+        onClickEditItem={(chain) => setDrilldownChain(chain.name)}
+        defaultSortAndFilterState={defaultSortAndFilterState}
+        ListComponent={ChainListItemWithCustom}
+        FilterComponent={ChainFilters}
+      />
+    );
+  }
+}
+
+function ChainListItem({
+  data: chain,
+  customField,
+}: {
+  data: ChainMetadata;
+  customField?: CustomListItemField;
+}) {
+  return (
+    <>
+      <div className="htw-flex htw-items-center">
+        <div className="htw-shrink-0">
+          <ChainLogo chainName={chain.name} logoUri={chain.logoURI} size={36} />
+        </div>
+        <div className="htw-ml-3 htw-text-left htw-shrink-0">
+          <div className="htw-text-sm htw-font-medium">{chain.displayName}</div>
+          <div className="htw-text-[0.7rem] htw-text-gray-500">
+            {chain.isTestnet ? 'Testnet' : 'Mainnet'}
+          </div>
+        </div>
+      </div>
+      <div className="htw-text-left">
+        <div className="htw-text-sm">
+          {customField
+            ? customField.data[chain.name] || 'Unknown'
+            : chain.deployer?.name || 'Unknown deployer'}
+        </div>
+        <div className="htw-text-[0.7rem] htw-text-gray-500">
+          {customField ? customField.header : 'Deployer'}
+        </div>
+      </div>
+    </>
+  );
+}
+
+function ChainFilters({
+  value,
+  onChange,
+}: {
+  value: ChainSortAndFilterState;
+  onChange: (s: ChainSortAndFilterState) => void;
+}) {
+  return (
+    <div className="htw-space-y-2">
+      <div className="htw-flex htw-items-center htw-justify-end htw-gap-1 sm:htw-gap-2">
+        <SegmentedControl
+          options={Object.values(ChainSortByOption)}
+          onChange={(selected) => onChange({ ...value, sortBy: selected! })}
+        />
+        <SegmentedControl
+          options={Object.values(SortOrderOption)}
+          onChange={(selected) => onChange({ ...value, sortOrder: selected! })}
+        />
+        <UpDownArrowsIcon
+          width={17}
+          height={17}
+          className="htw-hidden sm:htw-block"
+        />
+      </div>
+      <div className="htw-flex htw-items-center htw-justify-end htw-gap-1 sm:htw-gap-2">
+        <SegmentedControl
+          options={Object.values(FilterTestnetOption)}
+          onChange={(selected) =>
+            onChange({ ...value, filterTestnet: selected })
+          }
+          allowEmpty
+        />
+        <SegmentedControl
+          options={Object.values(ProtocolType)}
```

Comment:
unsure if we want this on a one-at-a-time toggle - might be useful to see EVM+Cosmos, excl sealevel or any other combination as we add more VMs e.g. fuel/starknet/move etc